UnitTests/PaymentsCoreTests/APIClientTests.swift

```diff
@@ -0,0 +1,50 @@
+import XCTest
+@testable import PaymentsCore
+
+class APIClientTests: XCTestCase {
+    
+    lazy var apiClient: APIClient = {
+        APIClient(
+            urlSession: URLSession(urlProtocol: URLProtocolMock.self),
+            environment: .sandbox
+        )
+    }()
+
+    func testAccessTokenMockRequest_Success() {
+        let expect = expectation(description: "Get mock response for access token request")
+
+        URLProtocolMock.requestResponses.append(MockAccessTokenRequestResponse(success: true))
```

Comment:
I think a lot of value of a good unit test suite comes from a test being readable in isolation. So instead of needing a tests to be super "DRY", I think there is more value in being able to go right to a failing test, read it and understand what that test is doing. So like in this example, the `"TestToken"` value is set in some other class through a few  different abstraction levels. I think this might be prematurely optimizing and that it would be easier to read this test and the mock JSON values all inline.

I think as the SDK & test suite scales, this introduces a bigger headache. So let's say a code change or iOS update causes a unit test to fail, now we also have to decipher how that set is set up across files vs just being able to read this one test in order to understand what it is doing.

Let me know if where I am coming from makes sense!

Reply:
I think it makes sense and that's a great point!!!! This is more like the setup we're having in PayPalCheckout and I agree it doesn't look nice asserting things from the mock in the test, but then having the mock object in another place.

What do you suggest when in the future we may use the same mock for multiple tests? Should we have a place dedicated for mocks, or will we agree with mocking similar responses in multiple tests?
I see it's the problem of "test readability" vs "clean code", but maybe for tests, readability should be higher priority? What do you think?

Reply:
I think this is a great point, and we could maybe forego adding default mock objects as statics, and instead create them locally from a response code / response data mock. maybe the lines would blur here if  a lot of tests want to use the same mock, but i don't think that should happen too often?

Reply:
We updated the init for our mocking types to include a response body string and a status code to be vended. We kept the access token response (and error sample) around as a static member of the access token response mock in order to prevent us from copying the same JSON block in multiple tests. How does this sound @scannillo ?

Reply:
We simplified the process for mocking requests - should be fairly easy to follow now. To create a mock response for a request:
```swift
        let mockRequest = MockRequestResponse(
            request: AccessTokenRequest(clientID: "123"), // this is any APIRequest 
            statusCode: 500, // this is the status code of the URL response
            responseString: nil, // this is the response data to vend (this should be a json string)
            error: serverError,  // this is the error object to vend
        )
```

This can be seen in the implementation of all our APIClient tests, and will prevent us from needing to create any special infrastructure for different request types while keeping the response easily discernible from the callsite in a test